src/main/java/UseCases/OtherAccountCases.java

```diff
@@ -1,19 +1,93 @@
 package UseCases;
 import Entities.User;
+import Entities.UserDataClasses.HideableUserDataClasses.Attributes;
+import Entities.UserDataClasses.HideableUserDataClasses.Interests;
 import Entities.UserDataClasses.UserData;
 import Entities.UserGraph;
+import Entities.UserDataClasses.HideableUserDataClasses.Course;
+import Entities.UserDataClasses.HideableUserDataClasses.Courses;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
 
 public class OtherAccountCases {
-    /*
     public String[] getUserInfo(String userid) {
+        try {
+            String[] userInfo = new String[6];
+            UserGraphReadWriter reader = new UserGraphReadWriter();
+            UserGraph graph = reader.readFromFile("userGraph.ser");
+            User userclass = graph.getUserByString(userid);
+
+            String displayName = userclass.getDisplayName().getData();
+            String location = userclass.getLocation().getData();
+            String bio = userclass.getBio().getData();
+
+            userInfo[0] = displayName;
+            userInfo[1] = location;
+            userInfo[2] = bio;
+
+            String courseString = "";
+            Courses courses = userclass.getCourses();
+            if (!courses.hidden) {
+                ArrayList<Course> courseList = courses.data;
+                for (int i = 0; i < courseList.size(); i++) {
+                    Course course = courseList.get(i);
+                    if (!course.hidden) {
+                        if (courseString.equals("")) {
+                            courseString += ", " + course.data;
+                        }
+                        else {
+                            courseString += course.data;
+                        }
+                    }
+                }
+            }
 
-        UserGraphReadWriter reader = new UserGraphReadWriter();
-        UserGraph graph = reader.readFromFile("dog.png");
-        User userclass = graph.getUserByString(userid);
-        String[] userInfo = {userclass.getDisplayName().getData(),
-                userclass.getLocation().getData(),
-                userclass.getBio().getData()};
-        return userInfo;
+            String interstsString = "";
+            Interests intersts = userclass.getInterests();
+            if (!intersts.hidden) {
+                HashMap<Integer, Boolean> map = intersts.data;
+                for (int i = 0; i <= 50; i++) {
```

Comment:
instead of doing i <=50, I would recommend using the size method on the interests dictionary, because that will allow us to add more interest in the future without having to come back and change this line. I had to do this earlier, just wrote the line:

```suggestion
                int numInterests = InterestsDict.interestMap.size();
                for (int i = 0; i < numInterests; i++) {
```